### PR TITLE
Modified GAME_START route and update CI/CD files

### PR DESCRIPTION
**Pull request by jzc1828**
- [x] Modify use of GAME_START route to auto play bonus tiles
- [x] Update dependencies
- [x] Modify eslint rules
- [x] Update CI file
- [x] Add CD file

See screenshot for result of this change:
image

### CODE REVIEW

File: src/pages/Game.tsx [hunk 1]
```diff
@@ -146,15 +147,43 @@ const GamePage = ({ ws, currentUser }: GameProps): JSX.Element => {
...
+      .getUsers()
+      .filter((user) => user.getConnectionId() !== mjPlayer.getConnectionId()) as MahjongOpponent[];
+    const opponentsBonusTilesObjs = selfPlayedTiles.filter(
+      (playedTiles) => playedTiles.connectionId !== mjPlayer.getConnectionId(),
+    );
+    const opponentsBonusTiles = opponentsBonusTilesObjs.map((playedTileObj: SelfPlayedTile) => {
+      const tilesArray = playedTileObj.playedTiles;
+      return tilesArray.map((strDef) => TileFactory.createTileFromStringDef(strDef));
+    });
+    mjOpponents.forEach((opponent, index) => opponent.addPlayedTiles(opponentsBonusTiles[index]));
```
Comment (yummyblabla) [inline src/pages/Game.tsx:186]: I see you're matching up the indices of the playedTiles with mjopponents. May be risky if states somehow changed. Maybe do 
opponentsBonusTiles.forEach((playedTileObj) =>{ find correct opponent in mjOpponent, and add played tiles }
Technically n^2, but if we care about efficiency (of n = 3 lol), we can always map it to {connectionId: MjOpponent }. Up to you.

Also, use opponent.addSelfPlayedTiles() addPlayedTiles is for creating melds from deadpile

File: src/pages/Game.tsx [hunk 2]
```diff
@@ -146,15 +147,43 @@ const GamePage = ({ ws, currentUser }: GameProps): JSX.Element => {
...
+    // Increment wall counter
+    const wallCounter = mjGameState.getWallCounter();
+    const result = wallCounter.setCurrentIndex(currentIndex);
+    if (!result) console.error('Failed to set wall counter. Please verify game state');
+
+    // Set player hand
     const tiles: Tile[] = [];
     tileArray.forEach((tile: string) => {
       tiles.push(TileFactory.createTileFromStringDef(tile));
     });
```
[the remark this answers is not in the document]
Comment (jzc1828) [reply src/pages/Game.tsx:163]: good to know u can do suggestion block 😄 
kk. Its ur code, so good refactoring!

File: src/pages/Game.tsx [hunk 3]
```diff
@@ -146,15 +147,43 @@ const GamePage = ({ ws, currentUser }: GameProps): JSX.Element => {
...
-
-    const mjPlayer = mjGameState.getMjPlayer();
+    const mjPlayer = mjGameState.getMjPlayer() as MahjongPlayer;
     mjPlayer.setHand(tiles);
+    const mjPlayerHand = mjPlayer.getHand();
+
+    // Set player played tiles (bonus tiles) at game start
+    const mjPlayerBonusTilesStrDef = selfPlayedTiles.filter(
+      (playedTiles: SelfPlayedTile) => playedTiles.connectionId === mjPlayer.getConnectionId(),
+    )[0].playedTiles;
```
[the remark this answers is not in the document]
Comment (jzc1828) [reply src/pages/Game.tsx:171]: kk

File: src/pages/Game.tsx [hunk 4]
```diff
@@ -146,15 +147,43 @@ const GamePage = ({ ws, currentUser }: GameProps): JSX.Element => {
...
+    const mjPlayerBonusTiles = mjPlayerBonusTilesStrDef.map((strDef) => TileFactory.createTileFromStringDef(strDef));
+    mjPlayerHand.addPlayedTiles(mjPlayerBonusTiles);
+
+    // Set opponents played tiles s tiles) at game start
```
[the remark this answers is not in the document]
Comment (jzc1828) [reply src/pages/Game.tsx:175]: kk

File: src/pages/Game.tsx (hunk 1 above)
Comment (jzc1828) [reply src/pages/Game.tsx:186]: `3 opponents ^ 2 = 9` vs `3`, not too bad haha

File: src/pages/Lobby.tsx [hunk 5]
```diff
@@ -311,7 +311,7 @@ const LobbyPage = ({ ws, currentUser }: LobbyProps): JSX.Element => {
         >
           <h2 className="">Games</h2>
           {games.map((game) => (
-            <GameRow gameId={game.gameId} onClickHandler={requestJoinGame} roomName={game.gameName} />
+            <GameRow gameId={game.gameId} onClickHandler={requestJoinGame} roomName={game.gameName} key={game.gameId} />
```
Comment (yummyblabla) [inline src/pages/Lobby.tsx:314]: Good catch. Saw this error on console but didn't really think too much of it

File: src/pages/Game.tsx [hunk 6]
```diff
@@ -146,15 +146,72 @@ const GamePage = ({ ws, currentUser }: GameProps): JSX.Element => {
...
+      const opponentHand = opponent.getHand();
+      const opponentPlayedTileObj = opponentsBonusTilesObjs.find(
+        (obj) => obj.connectionId === opponent.getConnectionId(),
+      );
+      if (opponentPlayedTileObj) {
+        const opponentPlayedTilesStrDef = opponentPlayedTileObj.playedTiles;
+        // eslint-disable-next-line
+        const opponentBonusTiles = opponentPlayedTilesStrDef.map((tile) => {
+          return TileFactory.createTileFromStringDef(tile);
+        });
```
Comment (yummyblabla) [inline src/pages/Game.tsx:199]: I'm assuming this violates line length ?
```suggestion
        const opponentBonusTiles = opponentPlayedTilesStrDef.map((tile) => TileFactory.createTileFromStringDef(tile));
```

File: src/pages/Lobby.tsx (hunk 5 above)
Comment (jzc1828) [reply src/pages/Lobby.tsx:314]: and this error actually introduced a small bug that u might see before:
user leaves a game by clicking on the Leave Game button, all the games in the game list are gone.

File: src/pages/Game.tsx (hunk 6 above)
Comment (jzc1828) [reply src/pages/Game.tsx:199]: can change to that, sometimes eslint is conflicting with prettier

**Pull request merged**